onnx/test/cpp/function_context_test.cc

```diff
@@ -0,0 +1,120 @@
+#include <iostream>
+#include "gtest/gtest.h"
+#include "onnx/checker.h"
+#include "onnx/common/constants.h"
+#include "onnx/defs/function.h"
+#include "onnx/defs/schema.h"
+
+using namespace ONNX_NAMESPACE::checker;
+
+namespace ONNX_NAMESPACE {
+namespace Test {
+
+// Utilities. TODO: Turn them into reusable ONNX utilities for use by
+
+TensorProto ToTensor(double value, TensorProto_DataType elem_type) {
```

Comment:
Can you use https://github.com/onnx/onnx/blob/master/onnx/defs/tensor_proto_util.cc#L10 ?

Reply:
No, this is a generalization of that. The difference is that this determines what to construct dynamically, based on the dynamic value of elem_type, while the other one has to be statically determined based on a static type.

Reply:
This is worth pushing into the other utility (as commented in the above code), but I don't think it is ready yet. It works for the test-case. But there are issues like how to convert a float into a float16 value which need to be resolved for a fully-general solution.